pandas/tests/arrays/test_datetimes.py

```diff
@@ -639,3 +641,47 @@ def test_tz_localize_t2d(self):
 
         roundtrip = expected.tz_localize("US/Pacific")
         tm.assert_datetime_array_equal(roundtrip, dta)
+
+    @pytest.mark.parametrize(
+        "error",
+        ["coerce", "raise"],
+    )
+    def test_fallback_different_formats(self, error):
+        # GH#46071
+        # 2 valid dates with different formats
+        # Should parse with no errors
+        s = pd.Series(["6/30/2025", "1 27 2024"])
+        expected = pd.Series(
+            [pd.Timestamp("2025-06-30 00:00:00"), pd.Timestamp("2024-01-27 00:00:00")]
+        )
+        result = pd.to_datetime(s, errors=error, infer_datetime_format=True)
+        tm.assert_series_equal(expected, result)
+
+    @pytest.mark.parametrize(
+        "dateseries",
+        [
+            pd.Series(["1/1/2000", "7/12/1200"]),
+            pd.Series(["1/1/2000", "Invalid input"]),
+        ],
+    )
+    def test_fallback_with_errors_coerce(self, dateseries):
+        # GH#46071
+        # Invalid inputs
+        # Parsing should fail for the second element
+        expected = pd.Series([pd.Timestamp("2000-01-01 00:00:00"), pd.NaT])
+        result = pd.to_datetime(dateseries, errors="coerce", infer_datetime_format=True)
+        tm.assert_series_equal(expected, result)
+
+    def test_fallback_with_errors_raise(self):
+        # GH#46071
+        # Invalid inputs
+        # Parsing should fail for the second element
+        dates1 = pd.Series(["1/1/2000", "7/12/1200"])
+        with pytest.raises(
+            OutOfBoundsDatetime, match="Out of bounds nanosecond timestamp"
+        ):
+            pd.to_datetime(dates1, errors="raise", infer_datetime_format=True)
```

Comment:
Isn't this duplicative of

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L961-L987

?